grind/bin/grind

```diff
@@ -439,11 +441,17 @@ class TestRunner:
         cmd = "%s batcharchive --dump-json=%s --"
         hashes_file = os.path.join(self.output_dir, "hashes.json")
         cmd = cmd % (self.config.isolate_path, hashes_file)
-        logger.debug("Invoking %s", cmd)
-        p = subprocess.Popen(shlex.split(cmd) + i.isolated_files, env=isolate_env)
-        p.wait()
-        if p.returncode != 0:
-            raise Exception("isolate batcharchive failed")
+        for num_retries_remaining in range(3, 0):
```

Comment:
Note that this produces an empty list. I think you want range(3, 0, -1).

I'd also prefer we do (3, -1, -1), that way it goes [3,2,1,0] and we can error when num_retries_remaining == 0.